src/main/java/com/minecolonies/coremod/entity/ai/citizen/enchanter/EntityAIWorkEnchanter.java

```diff
@@ -0,0 +1,379 @@
+package com.minecolonies.coremod.entity.ai.citizen.enchanter;
+
+import com.minecolonies.api.colony.ICitizenData;
+import com.minecolonies.api.colony.IColonyManager;
+import com.minecolonies.api.entity.ai.statemachine.AITarget;
+import com.minecolonies.api.entity.ai.statemachine.states.IAIState;
+import com.minecolonies.api.entity.citizen.AbstractEntityCitizen;
+import com.minecolonies.api.items.ModItems;
+import com.minecolonies.api.util.BlockPosUtil;
+import com.minecolonies.api.util.InventoryUtils;
+import com.minecolonies.coremod.MineColonies;
+import com.minecolonies.coremod.colony.buildings.AbstractBuildingWorker;
+import com.minecolonies.coremod.colony.buildings.workerbuildings.BuildingEnchanter;
+import com.minecolonies.coremod.colony.jobs.JobEnchanter;
+import com.minecolonies.coremod.entity.ai.basic.AbstractEntityAIInteract;
+import com.minecolonies.coremod.network.messages.CircleParticleEffectMessage;
+import com.minecolonies.coremod.network.messages.StreamParticleEffectMessage;
+import com.minecolonies.coremod.util.ExperienceUtils;
+import com.minecolonies.coremod.util.WorkerUtil;
+import net.minecraft.enchantment.Enchantment;
+import net.minecraft.enchantment.EnchantmentData;
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.item.ItemEnchantedBook;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.EnumParticleTypes;
+import net.minecraft.util.Tuple;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraftforge.items.wrapper.InvWrapper;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static com.minecolonies.api.entity.ai.statemachine.states.AIWorkerState.*;
+import static com.minecolonies.api.util.constant.Constants.TICKS_SECOND;
+import static com.minecolonies.api.util.constant.TranslationConstants.NO_WORKERS_TO_DRAIN_SET;
+
+/**
+ * Enchanter AI class.
+ */
+public class EntityAIWorkEnchanter extends AbstractEntityAIInteract<JobEnchanter>
+{
+    /**
+     * How often should intelligence factor into the enchanter's skill modifier.
+     */
+    private static final int INTELLIGENCE_MULTIPLIER = 2;
+
+    /**
+     * How often should intelligence factor into the enchanter's skill modifier.
+     */
+    private static final int CHARISMA_MULTIPLIER = 1;
+
+    /**
+     * Predicate to define an ancient tome which can be enchanted.
+     */
+    private static final Predicate<ItemStack> IS_ANCIENT_TOME = item -> !item.isEmpty() && item.getItem() == ModItems.ancientTome;
```

Comment:
We should probably change this to something else.
Refering only one item is painfull.
Either use a Tag or use a capability if additional data is needed.
However this is solved direct comparison to the item is not a viable solution anymore.

Reply:
I can make an instanceof too

Reply:
A tag is much better.
It falls within the nomenclature of modern mc development, as such please use a tag.
We should actually go much more in that direction.
Get rid of all those Config lists and replace with tags.

Reply:
Yeah but we only have tags in 1.14, in 1.12 we have to use oredict, which is not really nice

Reply:
I mean this would be a 1.14+ only feature for me.